### PR TITLE
Expose gRPC

### LINKED ISSUE #641: Expose gRPC
We have traditionally hidden the gRPC client and settings from the user, but the benefit of hiding gRPC no longer outweighs the cost of decreased user flexibility.

Two changes:

1. Add `WorkflowService() workflowservice.WorkflowServiceClient` to `Client`
2. Add `GRPCDialOptions []grpc.DialOption` to `ConnectionOptions` and apply our dial options first so any explicitly provided dial options _could_ override our own if desired.

Questions:

1. Maybe add some form of `GetGRPCContext` so that they can use the same context we use for making a gRPC call?

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

Added new `Client.WorkflowService` method that returns the underlying gRPC workflow service. Also expose gRPC dial options.

## Why?

People want to do advanced things that they cannot do with only our high-level wrapper (e.g. #570). This is exposed in other SDKs so should be here as well.

## Checklist

1. Closes #641

### CODE REVIEW

File: mocks/Client.go [hunk 1]
```diff
@@ -58,6 +57,11 @@ func (_m *Client) CancelWorkflow(ctx context.Context, workflowID string, runID s
 	return r0
 }
 
+// Close provides a mock function with given fields:
```
Comment (Sushisource) [inline mocks/Client.go:60]: This comment seems incomplete?

Comment (Sushisource) [reply mocks/Client.go:60]: Oh, nevermind I guess these are autogenerated? Weird.

**Review by Sushisource — APPROVED** [review]

File: mocks/Client.go (hunk 1 above)
Comment (cretz) [reply mocks/Client.go:60]: Yes, `mockery` is rubbish and were I to have built this project I'd have never used it (callers should use the mocking library of their choice, we should not provide mocks IMO).

Comment (Sushisource) [reply mocks/Client.go:60]: Yeah, I totally agree with that philosophy

**Pull request merged**